### DISCUSSION TITLE
Having a hard time understanding why its suggested to hide rendered poppers with CSS

### DISCUSSION THREAD

**Discussion by wisyr** [floating-ui/floating-ui#1293] [Q&A] [answered]
First of all thank for an outstanding library.

As in the title - Im having a hard time understanding why do you suggest to hide popper with applying css styles like opacity or visibility.

What if I have a list of thousands of elements that have a popper applied and I want to show the popper only when the element is hovered? The DOM tree is filled with thousands of poppers that are simply not visible, but rendered though - which negatively affects the performance.

Why the logic behind react-popper is not based on not rendering the popper at all if the element is not hovered/clicked? Just like e.g. material-ui.

Is there some way to solve it? How to not render elements that are not currently active? Thank you!

**Comment by FezVrasta** [comment, marked as answer]
Hi. You can unmount elements, but that means that when you render them again you must make sure to run a position update after the render is completed.
